File: core/kernel/runtime_policy.py
from __future__ import annotations

from typing import Any, Dict


def build_kernel_policy(
    max_phases: int = 20,
    max_graph_nodes: int = 1_000_000,
    require_kaalka_persistence: bool = True,
    allow_simulation: bool = True,
) -> Dict[str, Any]:
    return {
        "max_phases": max_phases,
        "max_graph_nodes": max_graph_nodes,
        "require_kaalka_persistence": require_kaalka_persistence,
        "allow_simulation": allow_simulation,
        "deterministic": True,
        "bounded": True,
    }


def enforce_kernel_policy(
    policy: Dict[str, Any],
    phase_count: int,
    node_count: int,
) -> Dict[str, Any]:
    within_phases = phase_count <= int(policy.get("max_phases", 20))
    within_graph = node_count <= int(policy.get("max_graph_nodes", 1_000_000))
    return {
        "allowed": within_phases and within_graph,
        "within_bounds": within_phases and within_graph,
        "bounded": True,
    }

File: src/kernel/runtimePolicy.ts
/**
 * Converted from Python: core/kernel/runtime_policy.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildKernelPolicy(max_phases: any = 20, max_graph_nodes: any = 1000000, require_kaalka_persistence: any = true, allow_simulation: any = true): any {
  return {"max_phases": max_phases, "max_graph_nodes": max_graph_nodes, "require_kaalka_persistence": require_kaalka_persistence, "allow_simulation": allow_simulation, "deterministic": true, "bounded": true};
}
export function enforceKernelPolicy(policy: any, phase_count: any, node_count: any): any {
  var within_phases: any = (phase_count <= py.toInt(py.get(policy, "max_phases", 20)));
  var within_graph: any = (node_count <= py.toInt(py.get(policy, "max_graph_nodes", 1000000)));
  return {"allowed": py.and2(within_phases, () => (within_graph)), "within_bounds": py.and2(within_phases, () => (within_graph)), "bounded": true};
}
